sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -120,31 +117,34 @@ class annotated_ref<T, detail::properties_t<Props...>> {
 #endif
   }
 
-  annotated_ref &operator=(const T &Obj) {
+  T operator=(const T &Obj) const {
 #ifdef __SYCL_DEVICE_ONLY__
     *__builtin_intel_sycl_ptr_annotation(
         m_Ptr, detail::PropertyMetaInfo<Props>::name...,
         detail::PropertyMetaInfo<Props>::value...) = Obj;
 #else
     *m_Ptr = Obj;
 #endif
-    return *this;
+    return Obj;
   }
 
-  annotated_ref &operator=(const annotated_ref &) = default;
+  T operator=(const annotated_ref &Ref) const { return *this = T(Ref); }
 
-  PROPAGATE_OP(+=)
-  PROPAGATE_OP(-=)
-  PROPAGATE_OP(*=)
-  PROPAGATE_OP(/=)
-  PROPAGATE_OP(%=)
-  PROPAGATE_OP(^=)
-  PROPAGATE_OP(&=)
-  PROPAGATE_OP(|=)
+  PROPAGATE_OP(+)
```

Comment:
We are removing these assign operators as we think these all need to go through the implicit conversion to have the annotation properly inserted. There is a PR here https://github.com/intel/llvm/pull/10770